vignettes/cpp11_faq.Rmd

```diff
@@ -0,0 +1,183 @@
+---
+title: "cpp11 FAQ"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{cpp11 FAQ}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+Intended to answer common questions about the usage of cpp11 objects
+
+## FAQ
+
+#### 1. How do I add elements to a named list?
+
+  * 
+  
+  ```{cpp11}
+  using namespace cpp11::literals;
+  [[cpp11::register]]
+  cpp11::list fn() {
+    cpp11::writable::list x;
+    x.push_back({"foo"_nm = 1});
+    return x;
+  }')
+  fn()
+  #> $foo
+  #> [1] 1
+  ```
+
+#### 2. How do I create a new empty environment?
+
+  ```
+  auto new_env = cpp11::package("base")["new.env"];
+  cpp11::environment my_env(new_env());
+  ```
+
+* How do I assign and retrieve values in an environment? What happens if I try to get a value that doesn't exist?
+
+  ```
+  my_env["foo"] = 1;
+  my_env["foo"] == 1;
+
+  my_env["bar"] == R_UnboundValue;
+  ```
+
+* How can I create a `cpp11:raws` from a `std::string`?
+
+  * Push back the contents individually
+  ```
+  cpp11::cpp_source(code = '#include <cpp11.hpp>
```

Comment:
You need to use the cpp11 knitr engine all the places you see `cpp_source()` in these examples.